src/main/java/org/jpeek/graph/XmlGraph.java

```diff
@@ -46,12 +51,15 @@
     /**
      * Ctor.
      * @param skeleton XMl representation on whiwh to build the graph
-     * @throws IOException If fails
+     * @param pname Package of the class this graph is for
+     * @param cname Class in the skeleton this graph is for
      */
-    public XmlGraph(final Skeleton skeleton) throws IOException {
+    public XmlGraph(
+        final Skeleton skeleton, final String pname, final String cname
+    ) {
         this.nds = new Unchecked<>(
             new Sticky<>(
-                () -> XmlGraph.build(skeleton)
+                () -> XmlGraph.build(skeleton, pname, cname)
```

Comment:
@vzurauskas why is it better to make `XmlGraph` for one class? Does it mean we need to introduce a todo to create another class responsible for iterating on all the classes of this skeleton?

Reply:
@victornoel `XmlGraph` implements this interface:

```java
public interface Graph {
    List<Node> nodes();
}
```
It's a list of nodes for one class, because if it was for all classes in skeleton, it would have to be a list of lists of nodes, or a map to lists of nodes. As far as I can tell, the graph is useful only in a context of a single class, because I don't think there are any cross-class cohesion metrics.

Iteration could be metric-specific, I guess. E.g. `Ccm` iterates through classes of a transformed skeleton and has a method (not implemented currently) to update the NCC value using a graph of a specific class.

Reply:
@vzurauskas ok, thank you for the explanation :)